scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -585,15 +585,9 @@
             ;; "Penalization" should only be employed when Fenzo does successfully match,
             ;; but the matches don't align with Cook's priorities.
             (empty? matches) true
-            (not (compare-and-set! pending-jobs scheduler-contents new-scheduler-contents))
```

Comment:
On the surface level, now we only have 2 branches in the cond - it might as well be an "if".

Going deeper, can you tell me a little more about this?  If we can just plow forward on atom contention rather than recycling, it's way simpler, but I don't know exactly why the contention handling was needed in the first place (what scenario it helped), so I can't feel safe declaring it's no longer needed.